packages/mobile/src/hooks/useLoadOfflineTracks.ts

```diff
@@ -10,24 +9,30 @@ import moment from 'moment'
 import { useDispatch } from 'react-redux'
 import { useAsync } from 'react-use'
 
-import { useFeatureFlag } from 'app/hooks/useRemoteConfig'
-import { loadTracks } from 'app/store/offline-downloads/slice'
+import { completeDownload, loadTracks } from 'app/store/offline-downloads/slice'
 
 import {
+  getOfflineCollections,
   getTrackJson,
   listTracks,
   verifyTrack
 } from '../services/offline-downloader/offline-storage'
 
+import { useIsOfflineModeEnabled } from './useIsOfflineModeEnabled'
+
 export const useLoadOfflineTracks = (collection: string) => {
-  const { isEnabled: isOfflineModeEnabled } = useFeatureFlag(
-    FeatureFlags.OFFLINE_MODE_ENABLED
-  )
+  const isOfflineModeEnabled = useIsOfflineModeEnabled()
+
   const dispatch = useDispatch()
 
   useAsync(async () => {
     if (!isOfflineModeEnabled) return
 
+    const offlineCollections = await getOfflineCollections()
+    offlineCollections?.forEach((collection) => {
+      dispatch(completeDownload(collection))
```

Comment:
Why do we dispatch `completeDownload` here?

Reply:
We're getting the stored list of collections marked for download. Calling `completeDownload` marks them as downloaded in state so we keep the toggle on across multiple sessions. We do this on completion of the track downloads for the collection, but next session we would forget.